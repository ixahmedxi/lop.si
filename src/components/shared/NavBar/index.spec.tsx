import { render } from 'test-utils'
import { NavBar, toggleMode } from '.'

describe('NavBar Component', () => {
  it('matches snapshot', () => {
    const { container } = render(<NavBar />)
    expect(container).toMatchSnapshot()
  })

  it('toggles theme', () => {
    const modes = ['dark', 'light']
    const mode = modes[0]
    const nextMode = toggleMode(mode, modes)

    expect(nextMode).toBe(modes[1])
  })
})
