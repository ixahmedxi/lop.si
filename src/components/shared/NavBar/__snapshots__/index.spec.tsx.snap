// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Component matches snapshot 1`] = `
.emotion-4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-top: 24px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media screen and (min-width:576px) {
  .emotion-4 {
    padding-top: 40px;
  }
}

.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-family: Pacifico,Arial,"Helvetica Neue",sans-serif;
  font-size: 24px;
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 24px;
  padding-right: 24px;
  padding-top: 16px;
  padding-bottom: 16px;
  color: white;
  background-color: var(--theme-ui-colors-primary,#8093f2);
  border: 0;
  border-radius: 4px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 40px;
  height: 40px;
  border-radius: 100px;
  margin-left: 16px;
  background-color: transparent;
  color: var(--theme-ui-colors-text,#EEEEEE);
  outline: none;
  cursor: pointer;
  padding: 0;
  box-shadow: none;
}

@media screen and (min-width:576px) {
  .emotion-1 {
    width: 40px;
    height: 40px;
    margin-left: 16px;
  }
}

@media screen and (min-width:768px) {
  .emotion-1 {
    width: 48px;
    height: 48px;
    margin-left: 16px;
  }
}

@media screen and (min-width:992px) {
  .emotion-1 {
    margin-left: 32px;
  }
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 40px;
  height: 40px;
  border-radius: 100px;
  margin-left: 16px;
  background-color: transparent;
  color: var(--theme-ui-colors-text,#EEEEEE);
  outline: none;
  cursor: pointer;
  padding: 0;
  box-shadow: none;
}

@media screen and (min-width:576px) {
  .emotion-2 {
    width: 40px;
    height: 40px;
    margin-left: 16px;
  }
}

@media screen and (min-width:768px) {
  .emotion-2 {
    width: 48px;
    height: 48px;
    margin-left: 16px;
  }
}

@media screen and (min-width:992px) {
  .emotion-2 {
    margin-left: 32px;
  }
}

<div>
  <nav
    class="emotion-4"
  >
    <a
      class="emotion-0"
    >
      lop.si
    </a>
    <div
      class="emotion-3"
    >
      <button
        aria-label="toggle theme"
        class="emotion-1"
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21 12.79A9 9 0 1 1 11.21 3 7 7 0 0 0 21 12.79z"
          />
        </svg>
      </button>
      <a
        aria-label="github repo"
        class="emotion-2"
        href="https://github.com/ixahmedxi/lop.si"
        rel="noopener noreferrer"
        target="_blank"
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9 19c-5 1.5-5-2.5-7-3m14 6v-3.87a3.37 3.37 0 0 0-.94-2.61c3.14-.35 6.44-1.54 6.44-7A5.44 5.44 0 0 0 20 4.77 5.07 5.07 0 0 0 19.91 1S18.73.65 16 2.48a13.38 13.38 0 0 0-7 0C6.27.65 5.09 1 5.09 1A5.07 5.07 0 0 0 5 4.77a5.44 5.44 0 0 0-1.5 3.78c0 5.42 3.3 6.61 6.44 7A3.37 3.37 0 0 0 9 18.13V22"
          />
        </svg>
      </a>
    </div>
  </nav>
</div>
`;
